### PR TITLE
Support CheckboxList descriptions

### PR DESCRIPTION
**Pull request by sumardi**
- [x] Changes have been thoroughly tested to not break existing functionality.
- [x] New functionality has been documented or existing documentation has been updated to reflect changes.
- [x] Visual changes are explained in the PR description using a screenshot/recording of before and after.

Just like radio `descriptions()`.

```php
Forms\Components\CheckboxList::make('methods')
    ->label(__('Payment methods'))
    ->options([
        'cash' => 'Cash',
        'transfer' => 'Bank transfer',
        'online' => 'Online banking',
        'credit' => 'Credit/debit card'
    ])
    ->descriptions([
        'cash' => 'Payment by cash.',
        'transfer' => 'Payment by bank transfer.',
        'online' => 'Payment by online banking.',
        'credit' => 'Pay by credit/debit card.',
    ])
    ->bulkToggleable()
    ->columns(2)
```

Screenshot 2023-06-09 at 5 23 48 PM
Screenshot 2023-06-09 at 5 23 36 PM

### CODE REVIEW

**Comment by zepfietje** [comment]
Thanks for the PR, @sumardi!

One note about the design: the checkbox input and label should still be centered vertically.
Could you make this change and mark as done when ready for review?

**Comment by sumardi** [comment]
> Thanks for the PR, @sumardi!
> 
> One note about the design: the checkbox input and label should still be centered vertically. Could you make this change and mark as done when ready for review?

Screenshot 2023-06-09 at 7 37 40 PM
Screenshot 2023-06-09 at 7 37 22 PM

Like this @zepfietje?

**Comment by zepfietje** [comment]
> Like this @zepfietje?

No, just the label, excluding description.
So something along those lines:
Screenshot 2023-06-09 at 17 15 44

**Comment by danharrin** [comment]
Hi @sumardi are you planning on completing this?

**Comment by zepfietje** [comment]
@sumardi, could you merge 2.x into this PR's branch to make sure it uses the formatter we added recently?

**Comment by sumardi** [comment]
> Hi @sumardi are you planning on completing this?

I am sorry that it took so long. Almost forgot that I've extended the class. I don't know if this is acceptable.

Screenshot 2023-06-28 at 6 27 24 PM

**Comment by sumardi** [comment]
> @sumardi, could you merge 2.x into this PR's branch to make sure it uses the formatter we added recently?

done.

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 1]
```diff
@@ -152,39 +152,53 @@ class="filament-forms-checkbox-list-component-option-label flex items-center spa
                             "
                         @endif
                     >
-                        <input
-                            @if ($isBulkToggleable())
+                        <div class="flex h-5 items-center">
+                            <input
+                                @if ($isBulkToggleable())
                                 x-on:change="checkIfAllCheckboxesAreChecked()"
                             @endif
```
Comment (zepfietje) [inline packages/forms/resources/views/components/checkbox-list.blade.php]:
```suggestion
                                    x-on:change="checkIfAllCheckboxesAreChecked()"
                                @endif
```

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 2]
```diff
@@ -152,39 +152,53 @@ class="filament-forms-checkbox-list-component-option-label flex items-center spa
...
+                                            'border-danger-300 ring-danger-500' => $errors->has($getStatePath()),
+                                            'dark:border-danger-400 dark:ring-danger-400' => $errors->has($getStatePath()) && config('forms.dark_mode'),
+                                        ])
+                                        ->merge([
+                                            'disabled' => $isDisabled(),
+                                        ])
+                                }}
+                            />
+                        </div>
+                        <div class="text-sm">
```
Comment (zepfietje) [inline packages/forms/resources/views/components/checkbox-list.blade.php:179]:
```suggestion

                        <div class="text-sm">
```

File: packages/forms/resources/views/components/checkbox-list.blade.php [hunk 3]
```diff
@@ -152,39 +152,54 @@ class="filament-forms-checkbox-list-component-option-label flex items-center spa
...
-                                        'text-primary-600 transition duration-75 rounded shadow-sm focus:border-primary-500 focus:ring-2 focus:ring-primary-500 disabled:opacity-70',
-                                        'dark:bg-gray-700 dark:checked:bg-primary-500' => config('forms.dark_mode'),
-                                        'border-gray-300' => ! $errors->has($getStatePath()),
-                                        'dark:border-gray-600' => (! $errors->has($getStatePath())) && config('forms.dark_mode'),
-                                        'border-danger-300 ring-danger-500' => $errors->has($getStatePath()),
-                                        'dark:border-danger-400 dark:ring-danger-400' => $errors->has($getStatePath()) && config('forms.dark_mode'),
-                                    ])
-                                    ->merge([
-                                        'disabled' => $isDisabled(),
+                        <div class="flex h-5 items-center">
```
Comment (zepfietje) [inline packages/forms/resources/views/components/checkbox-list.blade.php]: What is this fixed height for? 🙂

Comment (sumardi) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: > What is this fixed height for? 🙂

Without `h-5`
Screenshot 2023-06-28 at 7 16 11 PM

With `h-5`
Screenshot 2023-06-28 at 7 16 29 PM

😈

Comment (zepfietje) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: Hmm why is the label too heigh without `h-5` though?
In my non-Filament project that I showed earlier in this thread, I got it working without a fixed height, to allow for wrapping labels.

Comment (sumardi) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: > Hmm why is the label too heigh without `h-5` though? In my non-Filament project that I showed earlier in this thread, I got it working without a fixed height, to allow for wrapping labels.

Maybe you have input margin

Comment (zepfietje) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: Nope.

Edit: actually yes lol. `mt-0.5` I think that's the better approach.

Comment (sumardi) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: > Nope.
> 
> Edit: actually yes lol. `mt-0.5` I think that's the better approach.

Ok. thank you for your time.

Comment (sumardi) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: > Nope.
> 
> Edit: actually yes lol. `mt-0.5` I think that's the better approach.

`mt-1`
Screenshot 2023-06-28 at 10 07 57 PM

`mt-0.5`
Screenshot 2023-06-28 at 10 08 13 PM

I'm happy with `h-5` approach. 👻

Comment (zepfietje) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: Hmm neither looks centered. Could you figure out why?

Comment (sumardi) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: No

Comment (zepfietje) [reply packages/forms/resources/views/components/checkbox-list.blade.php]: Alright, then I'll have to take a look later.

**Comment by zepfietje** [comment]
We're retargeting this PR to 3.x instead of 2.x, @sumardi.
So it'll take a little more time before it gets merged.

**Pull request merged**